plugins/module_utils/ndi.py

```diff
@@ -0,0 +1,464 @@
+
+    def query_entry(self, path, size):
+        obj = self.nd.query_obj(path.format(size, 0), prefix=self.prefix)
+        entries = obj.get('entries')
+        if entries is None or len(entries) == 0:
+            return []
+        else:
+            pages, last_page = divmod(obj.get("totalItemsCount"), size)
+            if pages == 0 or (pages==1 and last_page==0) :
+                return entries
+            if last_page > 0 :
+                pages += 1
+            for page in range(1, pages):
+                obj = self.nd.query_obj(path.format(size, page), prefix = self.prefix)
+                entries += obj.get('entries')
+            return entries
+
+    def query_anomalies(self, ig_name, site_name, epoch_delta_job_id, epoch_choice):
+        epoch_delta_ig_path = self.epoch_delta_ig_path.format(ig_name, site_name, epoch_delta_job_id)
+        size = 100
+        path = epoch_delta_ig_path + "/individualTable?%24size={0}&%24page={1}"
+        if epoch_choice:
+            path = "{0}&epochStatus={1}".format(path, epoch_choice)
+        entries = self.query_entry(path, size)
+        return entries
+
+    def format_event_severity(self, events_severity):
+        result = {}
+        for each in events_severity:
+            event_severity_type = each.get("bucket").lower().split('_')[-1]
+            result[event_severity_type] = {}
+            for output in each.get("output"):
+                epoch = output.get("bucket").lower()
+                epoch_count = output.get("count")
+                result[event_severity_type][epoch] = epoch_count
+        return result
+
+    def format_impacted_resource(self, impacted_resource):
+        result = {}
+        for each in impacted_resource:
+            resource = each.get("bucket").lower()
+            result[resource] = {}
+            for output in each.get("output"):
+                for epoch in output.get("output"):
+                    epoch_type = epoch.get("bucket").lower()
+                    count = epoch.get("count")
+                    result[resource][epoch_type] = count
+        return result
+
+    def query_event_severity(self, ig_name, site_name, epoch_delta_job_id):
+        epoch_delta_ig_path = self.epoch_delta_ig_path.format(ig_name, site_name, epoch_delta_job_id)
+        path = "{0}/eventSeverity".format(epoch_delta_ig_path)
+        event_severity = self.query_data(path)
+        formated_event_severity = self.format_event_severity(event_severity)
+        return formated_event_severity
+
+    def query_impacted_resource(self, ig_name, site_name, epoch_delta_job_id):
+        epoch_delta_ig_path = self.epoch_delta_ig_path.format(ig_name, site_name, epoch_delta_job_id)
+        path = "{0}/impactedResource".format(epoch_delta_ig_path)
+        impacted_resource = self.query_data(path)
+        formated_impacted_resource = self.format_impacted_resource(impacted_resource)
+        return formated_impacted_resource
+
+    def format_messages(self, messages):
```

Comment:
self not used, should this be static method or function outside ndi class?

Reply:
will improve that later when refactoring codebase